### ISSUE TITLE
Question about Data Split in All Eval Datasets

### ISSUE THREAD

**Issue by Chen-X666** [Wuyxin/collabllm#22]
Awesome work! I saw that collabllm/collabllm-multiturn-math-hard only has a train split. Could you clarify how the data (MATH-Chat, Medium Doc Edit-Chat, Big Code Bench-Chat) is divided between training and evaluation?

**Comment by Wuyxin** [comment]
Hi, if you load the datasets without changing the seeds, the datasets will be the official splits we used in the paper.

**Comment by Chen-X666** [comment]
Thanks for your response😊! But actually, I’m a bit confused, since the paper doesn’t seem to clearly explain how the training and test sets were split. It only mentions 100 Medium articles as goal documents, 600 coding problems from BigCodeBench, and 200 level-5 math problems from MATH, which don’t seem to indicate a clear separation between training and evaluation data. Also, I noticed in the code that ds = MultiturnDataset(args.dataset_repo).to_inputs_dataset(eval_ratio=0.) suggests no separate evaluation set was created. Could you please clarify this point? Thanks a lot!

**Comment by Wuyxin** [comment]
hey, here's the split logic:

We use random split for medium and [bigcodebench](https://huggingface.co/datasets/bigcode/bigcodebench), and we follow the official split in https://huggingface.co/datasets/lighteval/MATH-Hard for math, but we only take 200 examples from the testing.

- We start with a single-turn dataset, with train and test split.
- When you run [build_dataset.py](https://github.com/Wuyxin/collabllm/blob/7378aab4efb1168557e72699cc51aebb5b8bc656/scripts/engine/build_dataset.py), it used only the **training set** to create a multiturn dataset repo. We leave the testing data as what it is.
- When you run online training code like [online_dpo.py](https://github.com/Wuyxin/collabllm/blob/7378aab4efb1168557e72699cc51aebb5b8bc656/scripts/train/online_dpo.py), it used the multiturn dataset repo -- `args.dataset_repo` with only the training data.

Particularly we set `eval_ratio=0.` in `ds = MultiturnDataset(args.dataset_repo).to_inputs_dataset(eval_ratio=0.)` because for online training, you don’t always need a separate validation set because the model’s performance is measured by the sequence of losses on the online training stream itself. And trl, if I remember correctly, doesn't log validation performance during the training. But of course, having a validation set doesn't hurt and can help, this is why we have validation set in our verl implementation. 

In short, `eval_ratio` here is just for train/val split, and we split the data to train(+val, optionally)/test in advance to this step. 
Hope this is helpful! Let me know if any question comes up!

**Comment by Chen-X666** [comment]
Thank you! These make sense now! Really appreciate your time and explanation. Have a nice day! 😊
